esmvalcore/preprocessor/_mask.py

```diff
@@ -298,27 +298,30 @@ def _mask_with_shp(cube, shapefilename, region_indices=None):
         raise ValueError(msg)
 
     # Wrap around longitude coordinate to match data
-    x_p_180 = np.where(x_p >= 180., x_p - 360., x_p)
+    x_p_180 = da.where(x_p >= 180., x_p - 360., x_p)
+
     # the NE mask has no points at x = -180 and y = +/-90
     # so we will fool it and apply the mask at (-179, -89, 89) instead
-    x_p_180 = np.where(x_p_180 == -180., x_p_180 + 1., x_p_180)
-    y_p_0 = np.where(y_p == -90., y_p + 1., y_p)
-    y_p_90 = np.where(y_p_0 == 90., y_p_0 - 1., y_p_0)
+    x_p_180 = da.where(x_p_180 == -180., x_p_180 + 1., x_p_180)
 
+    y_p_0 = da.where(y_p == -90., y_p + 1., y_p)
+    y_p_90 = da.where(y_p_0 == 90., y_p_0 - 1., y_p_0)
+
+    mask = None
     for region in regions:
         # Build mask with vectorization
-        if cube.ndim == 2:
+        if mask is None:
```

Comment:
why are you removing the check on the dimensionality? That's needed to see how to indez the `mask` array - is the new method taking care of that dynamically?

Reply:
Here, the mask is always 2D (lat, lon), so no need to check for the cube's dimensions. The `broadcast_to_shape` in l.319 will then make sure that this 2D mask is properly broadcasted to the cube shape.